tests/check_qwen_vision_encoder.py

```diff
@@ -0,0 +1,532 @@
+import unittest
+import jax
+import jax.numpy as jnp
+import numpy as np
+import torch
+import os
+
+from flax import nnx
+from jax.sharding import Mesh
+from transformers.models.qwen3_omni_moe.modeling_qwen3_omni_moe import (
+    Qwen3OmniMoeVisionPatchMerger as TorchQwen3OmniMoeVisionPatchMerger,
+    Qwen3OmniMoeVisionMLP as TorchQwen3OmniMoeVisionMLP,
+    Qwen3OmniMoeVisionPatchEmbed as TorchQwen3OmniMoeVisionPatchEmbed,
+    Qwen3OmniMoeVisionEncoder as TorchQwen3OmniMoeVisionEncoder,
+)
+
+from transformers.models.qwen3_omni_moe.modeling_qwen3_omni_moe import (
+    apply_rotary_pos_emb_vision,
+)
+from transformers.models.qwen3_omni_moe.configuration_qwen3_omni_moe import (
+    Qwen3OmniMoeVisionEncoderConfig,
+)
+from MaxText.layers.qwen3 import (
+    Qwen3OmniMoeVisionAttention as JaxQwen3OmniMoeVisionAttention,
+    Qwen3OmniMoeVisionPatchMerger as JaxQwen3OmniMoeVisionPatchMerger,
+    Qwen3OmniMoeVisionMLP as JaxQwen3OmniMoeVisionMLP,
+    Qwen3OmniMoeVisionPatchEmbed as JaxQwen3OmniMoeVisionPatchEmbed,
+    Qwen3OmniMoeVisionEncoder as JaxQwen3OmniMoeVisionEncoder,
+    Qwen3OmniMoeVisionProjector as JaxQwen3OmniMoeVisionProjector,
+)
+from MaxText.layers.embeddings import (
+    Qwen3OmniMoeVisionRotaryEmbedding as JaxQwen3OmniMoeVisionRotaryEmbedding,
+    Qwen3OmniMoeVisionPosEmbedInterpolate as JaxQwen3OmniMoeVisionPosEmbedInterpolate,
+)
+from MaxText import pyconfig
+from tests.multimodal_test_utils import (
+    copy_attention_weights_to_maxtext,
+    copy_patch_embed_weights,
+    copy_mlp_weights,
+    copy_patch_merger_weights,
+    copy_vision_encoder_weights,
+    create_random_jax_torch,
+    assert_all_close_jax_torch,
+    split_into_patches,
+)
+
+
+base_config_path = os.path.join(os.path.dirname(__file__), "..", "src", "MaxText", "configs", "base.yml")
+jax_vision_config = pyconfig.initialize(
```

Comment:
There are a few other precision related flag that may make the numerical gap smaller: `dtype_mm`, `float32_qk_product`, `activations_in_float32`. As used here: https://github.com/AI-Hypercomputer/maxtext/blob/main/tests/check_llama4_layers.py#L880-L892